src/server/pps/server/pipeline_controller.go

```diff
@@ -418,11 +394,11 @@ func (step *pcStep) rcIsFresh() bool {
 		log.Errorf("PPS master: RC for %q is nil", step.pipelineInfo.Pipeline.Name)
 		return false
 	}
-	expectedName := ""
-	if step.pipelineInfo != nil {
-		expectedName = ppsutil.PipelineRcName(step.pipelineInfo.Pipeline.Name, step.pipelineInfo.Version)
+	if step.pipelineInfo == nil {
```

Comment:
weird that this check was present previously - there's no situation where we should hit this, right?

Reply:
I know this is a drive-by comment, but I saw this and got curious: [here's the commit where this check was originally added](https://github.com/pachyderm/pachyderm/blob/8bacf0e1a62c555d190c4b53f4e7305a9bb62089/src/server/pps/server/pipeline_controller.go#L588), and as you can see, this code was totally different back then.

I think what I thought could happen was that the PPS master could read the `etcdPipelineInfo` from etcd, but then fail to read the full pipelineInfo from PPS for whatever reason, and then call `op.restartPipeline()`, which called `op.rcIsFresh()`, which would then crash without this check (though, reading through the code again, it seems like the pipeline should just fail in that case, so maybe even then it was unnecessary? I'm not totally sure, though :joy:)